### PR TITLE
HIP test case for 1D texture fetch

### PR DESCRIPTION
**Pull request by luckynikki**
Hip directed test cases for creating texture objects of linear and pitch type.

### CODE REVIEW

File: tests/src/texture/hipBindTexObj1D.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+bool testResult=true;
+
+__global__ void kernel(int *out)
+{
+
+	int x=blockIdx.x * blockDim.x + threadIdx.x;
+	out[x]=tex1Dfetch(tex,x);
+
+}
+
```
Comment (mangupta) [inline tests/src/texture/hipBindTexObj1D.cpp]: Please get rid of the additional empty lines.

File: tests/src/texture/hipBindTexObj1D.cpp [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        passed();
+    } else {
+        exit(EXIT_FAILURE);
+    }
+}
+
+void runTest()
+
+{
+string out;
```
Comment (mangupta) [inline tests/src/texture/hipBindTexObj1D.cpp]: Formatting is all wrong. Can you please run ```clang-format --style=file``` on this file? You can either use the clang-format in HCC bin directory or one that comes with clang/llvm.

File: tests/src/texture/tex1Dfetch_linear.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
+/* HIT_START
```
Comment (mangupta) [inline tests/src/texture/tex1Dfetch_linear.cpp]: Can't this test be combined into the above one i.e. into hipBindTexObj1D.cpp? Both seem to be working on 1d textures with fetch in linear mode.

File: tests/src/texture/hipBindTexObj1D.cpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
+/* HIT_START
```
Comment (mangupta) [inline tests/src/texture/hipBindTexObj1D.cpp]: Missing the copyright header.

File: tests/src/texture/hipBindTexObj1D.cpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+	  return;
+	}
+ }
+hipUnbindTexture(&tex);
+
+hipFree(tex_buf);
+
+
+}
+
```
Comment (mangupta) [inline tests/src/texture/hipBindTexObj1D.cpp]: Remove the extra empty lines from here as well.

**Comment by gargrahul** [comment]
@luckynikki @mangupta hipBindTexObj1D.cpp is testing nothing related to textures.
hipMalloc(&tex_buf, N * sizeof(int));
hipMemcpy(tex_buf, val, N * sizeof(int), hipMemcpyHostToDevice);
hipMemcpy(output, tex_buf, N * sizeof(int), hipMemcpyDeviceToHost);

Also passing a malloc pointer to kernel directly.

**Comment by gargrahul** [comment]
#478 and #481 fixed these tests.

**Pull request merged**